x/mint/keeper/emissions.go

```diff
@@ -74,30 +81,33 @@ func GetCirculatingSupply(
 	params types.Params,
 	blockHeight uint64,
 	blocksPerMonth uint64,
+	monthsAlreadyUnlocked math.Int,
 ) (
 	circulatingSupply,
 	totalSupply,
 	lockedVestingTokens,
 	ecosystemLocked math.Int,
+	updatedMonthsUnlocked math.Int,
 	err error,
 ) {
 	ecosystemBalance, err := k.GetEcosystemBalance(ctx, params.MintDenom)
 	if err != nil {
-		return math.Int{}, math.Int{}, math.Int{}, math.Int{}, err
+		return math.Int{}, math.Int{}, math.Int{}, math.Int{}, math.Int{}, err
 	}
 	totalSupply = params.MaxSupply
-	lockedVestingTokens, _, _, _ = GetLockedVestingTokens(
+	lockedVestingTokens, _, _, _, updatedMonthsUnlocked = GetLockedVestingTokens(
 		blocksPerMonth,
 		math.NewIntFromUint64(blockHeight),
 		params,
+		monthsAlreadyUnlocked.ToLegacyDec(),
```

Comment:
the concepts of `monthsAlreadyUnlocked` and `updatedMonthsUnlocked` seems so similar, yet are manipulated in 2 different scopes. for this, I'm not going to block, as there is some overlapping control flow (to calc updatedMonthsUnlocked and the other values, obvi) but I think it'd be cleaner to separate this logic from the compute of vested tokens